src/engine/effects/engineeffectsdelay.cpp

```diff
@@ -0,0 +1,105 @@
+#include "engine/effects/engineeffectsdelay.h"
+
+#include "util/sample.h"
+
+namespace {
+// See enginedelay.cpp
+const int kiMaxDelay = static_cast<int>(0.508 *
+        mixxx::audio::SampleRate::kValueMax * mixxx::kEngineChannelCount);
```

Comment:
We have also no definition for a i prefix. 0.508 is a magic number can we name it? What is the unit of kMaxDelay?

Reply:
We should refine this anyways. Also rounding up to a "power of two - 1" helps the optimizer because the expensive modulo operation can be transformed into a binary AND.

Reply:
> We have also no definition for a i prefix. 0.508 is a magic number can we name it? What is the unit of kMaxDelay?

I agree that is a magic number. The value was used from the [EngineDelay](https://github.com/mixxxdj/mixxx/blob/bf98de330910c38bd9b3cf51fb50dbe0188f4690/src/engine/enginedelay.cpp#L11) using maximum delay pot. The value can be changed.

Reply:
> We should refine this anyways. Also rounding up to the next power of 2 helps the optimizer because the expensive modulo operation can be transformed into a binary AND.

I agree. That's a good point, thank you.

Reply:
So, just for summarizing an easy calculation. Following the current calculation, with magic constant 0.508 from `enginedelay.cpp` and using `mixxx::audio::SampleRate::kValueMax`(which is 192 000) and `mixxx::kEngineChannelCount` (which is stereo), the following power of two is 2^18 (262 144) or next 2^19 (524 288). I am thinking about whether it makes sense to think about a greater buffer because the following power of two (2^20) is over 1 million.

Reply:
Mhmmm. The fact that we need to account for the (unlikely) worstcase (192Khz samplingrate) makes this calculation somewhat difficult. I wish we could allocate the buffer based on the actual buffer size currently being used instead of the worst case.
What do we think is an acceptable maximum delay factor? 2x the current buffer size?